bpf/lib/policy.h

```diff
@@ -71,7 +71,19 @@ policy_sk_egress(__u32 identity, __u32 ip,  __u16 dport)
 		return CTX_ACT_OK;
 	}
 
+	/* Check Proto-only policy */
+	key.sec_label = 0;
+	policy = map_lookup_elem(map, &key);
+	if (likely(policy)) {					/* 4. ANY/proto/ANY */
+		/* FIXME: Need byte counter */
+		__sync_fetch_and_add(&policy->packets, 1);
+		if (unlikely(policy->deny))
+			return DROP_POLICY_DENY;
+		return CTX_ACT_OK;
+	}
```

Comment:
It'd be nice to avoid these extra lookups if we don't have such policies or something. Even on the original PR that introduced this support we had a bit of discussion around the performance impact, and now we're adding yet more checks to the policy code for every packet. That said, I don't think such a change is appropriate for this PR. I'll defer to @cilium/sig-datapath on how much they want to pursue performance optimization on this front.

Reply:
Should not we create an issue for that? At least to keep track of your point and perhaps if some contributors have interest, open the discussion...

Reply:
Looking back, while I raised this concern previously, I didn't see any interest raised from sig-datapath folks. I don't know for sure what the impact of this particular set of changes is vs. the rest of the datapath, so maybe it's a bit premature to raise an issue specifically for this. It probably makes more sense that we look at performance more holistically and just assess performance in general. As part of that, we can follow up wherever the low-hanging fruit are. We can certainly file an issue for "investigating network performance bottlenecks in Cilium", but unless someone is actively interested and looking into it, then I'm not sure whether it matters to raise the issue now. When someone's interested they can file an issue and we can discuss the findings.